src/machine/machine_atmega.go

```diff
@@ -155,3 +155,91 @@ func (uart UART) WriteByte(c byte) error {
 	avr.UDR0.Set(c) // send char
 	return nil
 }
+
+// SPIConfig is used to store config info for SPI.
+type SPIConfig struct {
+	Frequency uint32
+	SCK       Pin
+	SDO       Pin
+	SDI       Pin
+	LSBFirst  bool
+	Mode      uint8
+}
+
+type SPI struct{}
+
+// Configure is intended to setup the SPI interface.
+func (spi SPI) Configure(config SPIConfig) error {
+
+	// Make the defaults meaningful
+	if config.Frequency == 0 {
+		config.Frequency = 4000000
+	}
+	if 0 == config.SCK {
```

Comment:
You might want to resolve the Yoda Conditions, by switching the order of the check. 

Example: config.SCK == 0  instead of 0 == config.SCK